web/src/engine/osk/src/input/gestures/browser/oskSubKey.ts

```diff
@@ -24,7 +25,10 @@ export default class OSKSubKey extends OSKKey {
     let ks=kDiv.style;
 
     for(var tp in tKey) {
-      if(typeof spec[tp] != 'string') {
+      // We've already preprocessed the keyboard's version of the subkey.  While certain
+      // layout properties are fine to overwrite, certain functional properties must
+      // be preserved.
+      if(typeof spec[tp] != 'string' && tp != 'default' && tp != '_baseKeyEvent') {
         spec[tp]=tKey[tp];
       }
     }
```

Comment:
I find this codeblock as originally written to be kind of suspect.  If the base spec for a key isn't type string... overwrite specific values _with the default_?  Personally, I want to change it to something that seems better-fitting, but I figured that I'd keep the scope of changes minimal.

Getting rid of the former intermediate `OSKKeySpec` means that the keyboard spec object for subkeys is now exposed directly here, and both 'default' (for default subkeys) and the `baseKeyEvent` object aren't type `string`.  Those _must not_ be overwritten.  These always had direct references on the base key, but the subkeys previously made a pseudo-clone for their constructed DOM equivalents - the `OSKKeySpec` instances - that shielded the actual spec from mutations.

I'm actually not 100% sure at the moment how those were previously being retrieved when applied,

Reply:
> If the base spec for a key isn't type string... overwrite specific values _with the default_?

Reasons lost in the deep dark mists of time but I suspect it was to do with ensuring valid types for properties later on in usage -- I guess it considered any prop that wasn't already a string to be invalid. And so we may need to be careful as properties have non-string types now, right?

Reply:
> And so we may need to be careful as properties have non-string types now, right?

Yep, that's why I'm raising the concern.  We've long noted how certain integer types were being coerced into strings - dropping that coercion means we now 'get' to have non-string types, and that matters greatly at this specific location.

Reply:
Do we need to drop this 'validation' altogether? It seems wrong to me at this point...

Reply:
My proposal for this is at #9804.  I felt it'd be wise to get that properly reviewed, and that's naturally the easiest when changes are in isolation.